app/config/config.yml

```diff
@@ -66,3 +66,15 @@ swiftmailer:
     username:  "%mailer_user%"
     password:  "%mailer_password%"
     spool:     { type: memory }
+
+# Can be changed directly in the bundle is we found an agreement
+dunglas_action:
+    autodiscover:
+        enabled:   true
+        directories:
+            action: [ ]
+            console: [ Command ]
+    directories:         # List of directories relative to the kernel root directory containing classes to auto-register.
+        action: [ '../src/AppBundle/Controller' ] # Still an autowiring_type definition to add in Symfony to avoid this
```

Comment:
Yea, I see now that `action-annotation` does nothing more than auto-set the `_controller` to the `action.*` pattern. That feels weak.

A more generic solution where `@Route` just *works* with service controllers looks even more appealing. In fact, if we could pull it off, then it could work in YAML too using the existing syntax - e.g.

```yml
homepage:
    path: /
    defaults:
        # or using the A:B:C syntax
       _controllers: AppBundle\Controller\MainController::homepage
```

If that *class* were represented as a service, it could/would use that service automatically. Otherwise, it would create a new instance like now.

I'm assuming you've thought about this - it would require a service that contained a run-time class => service id map for resolution.